File: string/Solution58.java
class Solution58 {
    public int lengthOfLastWord(String s) {
    	int x = 0;
        boolean collected = false;
    	for(int i = s.length() - 1; i >= 0; i--){
    		char cur = s.charAt(i);
    		if(cur != ' '){
    			x += 1;
    			collected = true;
    		}
    		else if(collected){
                return x;
            }
    	}
    	return x;
    }

    public static void main(String[] args){
    	Solution58 s = new Solution58();
    	System.out.println(s.lengthOfLastWord("a "));
    }
}

File: C++/Solution58.cpp
#include<iostream>
#include<vector>
#include<algorithm>
#include<unordered_map>
#include<unordered_set>
using namespace std;

class Solution58 {
public:
    int lengthOfLastWord(string s) {
        int len = 0;
        bool collect = false;
        for(int index = s.length() - 1; index >= 0; index--){
            if(s[index] != ' '){
                len++;
                collect = true;
            }
            else{
                if(collect){
                    return len;
                }
            }
        }
        return len;
    }
};